sway-core/src/language/ty/expression/expression.rs

```diff
@@ -70,8 +70,10 @@ impl CollectTypesMetadata for TyExpression {
             FunctionApplication {
                 arguments,
                 function_decl_id,
+                call_path,
                 ..
             } => {
+                ctx.set_call_site_span(Some(call_path.span()));
```

Comment:
Do we need to do this for every possible expression type in this `match` statement?

Reply:
This may be needed for other expressions, I should write test cases for the other expressions.

Reply:
Added test case and fix for when it cannot infer struct parameter types. There may exist more cases that I didn't think of. But for now it already fixes 3 uses cases.